src/js/common/schemaform/FormPage.jsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { withRouter } from 'react-router';
+import _ from 'lodash/fp';
+import Scroll from 'react-scroll';
+import Form from 'react-jsonschema-form';
+import { uiSchemaValidate, transformErrors } from './validation';
+import FieldTemplate from './FieldTemplate';
+import * as widgets from './widgets';
+import ProgressButton from '../components/form-elements/ProgressButton';
+
+import { focusElement } from '../utils/helpers';
+
+const fields = {
+};
+
+const scrollToFirstError = () => {
+  setTimeout(() => {
+    const errorEl = document.querySelector('.usa-input-error, .input-error-date');
+    if (errorEl) {
+      const position = errorEl.getBoundingClientRect().top + document.body.scrollTop;
+      Scroll.animateScroll.scrollTo(position - 10, {
+        duration: 500,
+        delay: 0,
+        smooth: true
+      });
+      focusElement(errorEl);
+    }
+  }, 100);
+};
+const scroller = Scroll.scroller;
+
+const scrollToTop = () => {
+  scroller.scrollTo('topScrollElement', {
+    duration: 500,
+    delay: 0,
+    smooth: true,
+  });
+};
+
+/*
+ * Each page uses this component and passes in config. This is where most of the page level
+ * form logic should live.
+ */
+class FormPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.validate = this.validate.bind(this);
+    this.onBlur = this.onBlur.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onError = this.onError.bind(this);
+    this.goBack = this.goBack.bind(this);
+    this.transformErrors = this.transformErrors.bind(this);
+    this.state = this.getEmptyState(props.route.pageConfig);
+  }
+  componentDidMount() {
+    scrollToTop();
+  }
+  componentWillReceiveProps(newProps) {
+    if (newProps.route.pageConfig !== this.props.route.pageConfig) {
+      this.setState(this.getEmptyState(newProps.route.pageConfig));
+    }
+  }
+  componentDidUpdate(prevProps) {
+    if (prevProps.route.pageConfig !== this.props.route.pageConfig) {
+      scrollToTop();
+    }
+  }
+  onBlur(id) {
+    const formContext = _.set(['touched', id], true, this.state.formContext);
+    this.setState({ formContext });
+  }
+  onChange({ formData }) {
+    this.setState({ formData });
+  }
+  onError() {
+    const formContext = _.set('submitted', true, this.state.formContext);
+    this.setState({ formContext });
+    scrollToFirstError();
+  }
+  onSubmit() {
+    const { pageList, pageConfig } = this.props.route;
+    const pageIndex = _.findIndex(item => item.pageKey === pageConfig.pageKey, pageList);
+    this.props.router.push(pageList[pageIndex + 1].path);
+  }
+  getEmptyState(pageConfig) {
+    return { formData: pageConfig.initialData, formContext: { touched: {}, submitted: false } };
+  }
+  goBack() {
+    const { pageList, pageConfig } = this.props.route;
+    const pageIndex = _.findIndex(item => item.pageKey === pageConfig.pageKey, pageList);
+    this.props.router.push(pageList[pageIndex - 1].path);
```

Comment:
If you're going back to another page in the form, you should get scrolled to the top in the `componentDidUpdate` method. We may want to add a scroll to the Intro page, though.